carma-messenger-core/cpp_message/src/SDSM_Message.cpp

```diff
@@ -1072,4 +1072,958 @@ namespace cpp_message
 
     }
 
+
+
+    // SDSM decoding
+    boost::optional<j3224_v2x_msgs::msg::SensorDataSharingMessage> SDSM_Message::decode_sdsm_message(std::vector<uint8_t>&binary_array){
+        
+        j3224_v2x_msgs::msg::SensorDataSharingMessage output;
+        // decode results - stored in binary_array
+        asn_dec_rval_t rval;
+        MessageFrame_t* message = nullptr;
+        
+        // copy from vector to array         
+        size_t len = binary_array.size();    
+        uint8_t buf[len];             
+        std::copy(binary_array.begin(),binary_array.end(),buf);
+        
+        // use asn1c lib to decode
+        rval = uper_decode(0, &asn_DEF_MessageFrame,(void **) &message, buf, len, 0, 0);
+
+        if(rval.code == RC_OK)
+        {
+            // Incoming SDSM in ASN.1 C-struct format
+            SensorDataSharingMessage_t sdsm_core = message->value.choice.SensorDataSharingMessage;
+            
+            // MessageCount
+            if(sdsm_core.msgCnt < j2735_v2x_msgs::msg::MsgCount::MSG_COUNT_MAX){
+                output.msg_cnt.msg_cnt = sdsm_core.msgCnt;
+                std::cout << sdsm_core.msgCnt << std::endl;
```

Comment:
We should be using ROS's logging functionality instead of outputting to the standard output directly.

Reply:
debug statement removed